go/controller/internal/controller/gitrepository_controller.go

```diff
@@ -144,18 +143,13 @@ func (r *GitRepositoryReconciler) handleDelete(ctx context.Context, repo *v1alph
 	logger := log.FromContext(ctx)
 	if controllerutil.ContainsFinalizer(repo, RepoFinalizer) {
 		logger.Info("delete git repository")
-		if repo.Status.ID == nil {
```

Comment:
This was causing repository to get stuck in a delete loop if it was deleted before ID was synchronized.